src/test/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessorTest.java

```diff
@@ -192,38 +205,75 @@ public void process() throws Exception {
         DownloadParticipantRosterRequest request = new DownloadParticipantRosterRequest.Builder()
                 .withUserId(USER_ID).withAppId(APP_ID).withPassword(PASSWORD).build();
 
-        // set up mocks
+        // mock caller
         doReturn(participant).when(mockBridgeHelper).getParticipant(APP_ID, USER_ID, false);
 
-        List<StudyParticipant> studyParticipants = ImmutableList.of(makeStudyParticipant(1), makeStudyParticipant(2), makeStudyParticipant(3));
+        // mock multiple bridge calls to test pagination
+        StudyParticipant participant1 = makeStudyParticipant(1);
+        StudyParticipant participant2 = makeStudyParticipant(2);
+        StudyParticipant participant3 = makeStudyParticipant(3);
+
+        doReturn(ImmutableList.of(participant1)).when(mockBridgeHelper).getStudyParticipantsForApp(APP_ID, ORG_ID, 0, 1, null);
+        doReturn(ImmutableList.of(participant2)).when(mockBridgeHelper).getStudyParticipantsForApp(APP_ID, ORG_ID, 1, 1, null);
+        doReturn(ImmutableList.of(participant3)).when(mockBridgeHelper).getStudyParticipantsForApp(APP_ID, ORG_ID, 2, 1, null);
+
+        AppInfo appInfo = new AppInfo.Builder().withAppId(APP_ID).withName("test-name").withSupportEmail("example@example.org").build();
+        doReturn(appInfo).when(mockDynamoHelper).getApp(APP_ID);
 
-        doReturn(studyParticipants).doReturn(ImmutableList.of()).when(mockBridgeHelper).getStudyParticipantsForApp(eq(APP_ID), eq(ORG_ID), eq(0), eq(PAGE_SIZE), anyString());
+        AccountInfo accountInfo = new AccountInfo.Builder().withUserId(USER_ID).withEmailAddress("example@example.org").build();
+        doReturn(accountInfo).when(mockBridgeHelper).getAccountInfo(APP_ID, USER_ID);
 
         // we want to evaluate the files before they're cleaned up
         doNothing().when(processor).cleanupFiles(csvFile, zipFile, tmpDir);
 
         // process
-        processor.process(request, csvFile, zipFile);
+        processor.setPageSize(1); // the processor will need to make 3 Bridge API calls
+        processor.process(request, csvFile, zipFile, tmpDir);
 
         // verify file contents
         String csvContent = new String(fileHelper.getBytes(csvFile), StandardCharsets.UTF_8);
         assertEquals(EXPECTED_CSV_CONTENT, csvContent);
 
         // verify that the csv is written to the file
-        verify(processor).writeStudyParticipants(any(CSVWriter.class), eq(studyParticipants), eq(0), eq(APP_ID), eq(ORG_ID), anyString());
+        verify(processor).writeStudyParticipants(any(CSVWriter.class), anyList(), eq(0), eq(APP_ID), eq(ORG_ID), eq(null), anyInt());
```

Comment:
I think there's some confusion here as to why we're calling verify() on the method that we're testing.

Reply:
I think I understand what you're saying - I just replaced anyList() with an ImmutableList.of(participant1), which constitutes the first "page" of study participants to be written. Now the params are being checked.